nixos/modules/services/misc/tzupdate.nix

```diff
@@ -25,6 +25,7 @@ in {
     # a service to run after you have *internet* access.
     systemd.services.tzupdate = {
       description = "tzupdate timezone update service";
+      wantedBy = [ "multi-user.target" ];
```

Comment:
The comment above indicates that this isn’t done because we don’t (or might not?) have Internet access at this point.

I think it’s reasonable to run at startup anyway, and have the service fail if there’s no Internet (or does it just succeed without changing the time?), but the comment should be changed if so.

Reply:
That still holds true, as you haven't removed the `after` options :)

Reply:
@xanderio I’m not sure what you’re referring to.

I’m suggesting changing
```nix
    # We provide a one-shot service which can be manually run. We could
    # provide a service that runs on startup, but it's tricky to get
    # a service to run after you have *internet* access.
```
to something like, “We provide a one-shot service that runs at startup once network interfaces are up, but we can’t ensure we actually have *Internet* access at that point. It can also be run manually.”

I don’t think the `after` options affect any part of the existing comment.

Reply:
Let's just say I've wrote that before me first tea of the day ^^'.

Reply:
I was trying to have the change minimal. However, https://github.com/NixOS/nixpkgs/issues/127984#issuecomment-2511448325 suggests setting up a timer for the service.

IMHO that's a good approach too. It can be done in a new option. Sometimes, you don't actually reboot your laptop when you cross the border, and still you'd want the timezone to be updated at some point. That also happens with phones.

So, maybe instead of telling people to run manually a service, we could tell them that theres this other option (let's say `services.tzupdate.timer.enable = true` that they can use to let tzupdate run under some schedule.

So, do you think it makes sense to add the timer too?

Reply:
I am for adding the timer. I already set up the timer manually myself, and it seems to add no configuration complexity to just have `services.tzupdate.interval = null` (or whatever the name is) as the default.